pytest_fixtures/satellite_auth.py

```diff
@@ -0,0 +1,153 @@
+import copy
+
+from nailgun import entities
+from pytest import fixture
+
+from robottelo.config import settings
+from robottelo.constants import AUDIENCE_MAPPER
+from robottelo.constants import GROUP_MEMBERSHIP_MAPPER
+from robottelo.constants import LDAP_ATTR
+from robottelo.constants import LDAP_SERVER_TYPE
+from robottelo.datafactory import gen_string
+from robottelo.rhsso_utils import create_mapper
+from robottelo.rhsso_utils import get_rhsso_client_id
+from robottelo.rhsso_utils import run_command
+
+
+@fixture(scope='function')
+def auth_source(module_org, module_loc):
+    ldap_data = {
+        'ldap_user_name': settings.ldap.username,
+        'ldap_user_passwd': settings.ldap.password,
+        'base_dn': settings.ldap.basedn,
+        'group_base_dn': settings.ldap.grpbasedn,
+        'ldap_hostname': settings.ldap.hostname,
+    }
+    return entities.AuthSourceLDAP(
+        onthefly_register=True,
+        account=ldap_data['ldap_user_name'],
+        account_password=ldap_data['ldap_user_passwd'],
+        base_dn=ldap_data['base_dn'],
+        groups_base=ldap_data['group_base_dn'],
+        attr_firstname=LDAP_ATTR['firstname'],
+        attr_lastname=LDAP_ATTR['surname'],
+        attr_login=LDAP_ATTR['login_ad'],
+        server_type=LDAP_SERVER_TYPE['API']['ad'],
+        attr_mail=LDAP_ATTR['mail'],
+        name=gen_string('alpha'),
+        host=ldap_data['ldap_hostname'],
+        tls=False,
+        port='389',
+        organization=[module_org],
+        location=[module_loc],
+    ).create()
+
+
+@fixture(scope='function')
+def auth_source_ipa(module_org, module_loc):
+    ipa_data = {
+        'ldap_ipa_user_name': settings.ipa.username_ipa,
+        'ipa_otp_username': settings.ipa.otp_user,
+        'ldap_ipa_user_passwd': settings.ipa.password_ipa,
+        'ipa_base_dn': settings.ipa.basedn_ipa,
+        'ipa_group_base_dn': settings.ipa.grpbasedn_ipa,
+        'ldap_ipa_hostname': settings.ipa.hostname_ipa,
+        'time_based_secret': settings.ipa.time_based_secret,
+    }
+    return entities.AuthSourceLDAP(
+        onthefly_register=True,
+        account=ipa_data['ldap_ipa_user_name'],
+        account_password=ipa_data['ldap_ipa_user_passwd'],
+        base_dn=ipa_data['ipa_base_dn'],
+        groups_base=ipa_data['ipa_group_base_dn'],
+        attr_firstname=LDAP_ATTR['firstname'],
+        attr_lastname=LDAP_ATTR['surname'],
+        attr_login=LDAP_ATTR['login'],
+        server_type=LDAP_SERVER_TYPE['API']['ipa'],
+        attr_mail=LDAP_ATTR['mail'],
+        name=gen_string('alpha'),
+        host=ipa_data['ldap_ipa_hostname'],
+        tls=False,
+        port='389',
+        organization=[module_org],
+        location=[module_loc],
+    ).create()
+
+
+@fixture(scope='session')
+def ldap_auth_name():
+    """Return some random ldap name, and attempt to delete all ldap when test starts.
```

Comment:
I think this fixture could use a more relevant name to the actions that its taking.

In fact it appears to me to be two separate fixtures, one of which is questionable as a fixture:

1. `def delete_auth_source_users`

A fixture that deletes the users related to auth sources - does this need to be executed for each test that uses the  `auth_source` fixture?  Why only do this at the start of the test session?


2. Random alphanumeric string

This doesn't need to be a fixture, it looks like each of the tests is using this to then call a `create` - that random string can just be called within the test function.  IMO fixtures should be taking _actions_ to setup, not just providing randomized data or data from settings.

Reply:
> I think this fixture could use a more relevant name to the actions that its taking.
> 
> In fact it appears to me to be two separate fixtures, one of which is questionable as a fixture:
> 
> 1. `def delete_auth_source_users`
> 
> A fixture that deletes the users related to auth sources - does this need to be executed for each test that uses the `auth_source` fixture? Why only do this at the start of the test session?

Yeah, it was old code, Maybe It was started as the LDAP username after a while it was updated the function. I can update the name make sense to me.   
> 1. Random alphanumeric string
> 
> This doesn't need to be a fixture, it looks like each of the tests is using this to then call a `create` - that random string can just be called within the test function. IMO fixtures should be taking _actions_ to setup, not just providing randomized data or data from settings.

This is because we ldap_auth and users generated by the tier 2 tests from API or CLI can cause the data issue. To make sure that we wanted to have a clean set up before going further on UI tests